### ISSUE TITLE
[no-implicit-any-catch] the rule isn't listed anymore in the README

### ISSUE THREAD

**Issue by nstepien** [typescript-eslint/typescript-eslint#4060]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

- Open https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md
- Check the rules listed
- Compare with the list of rules docs
  - https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/docs/rules
- Compare with the list of rules source
  - https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/rules

```JSON
{}
```

```TS
// your repro code case
```

**Expected Result**

`no-implicit-any-catch` should be listed in the README.

**Actual Result**

`no-implicit-any-catch` is not listed in the README.

**Additional Info**

There may be more discrepancies, I haven't checked everything, just noticed this when updating from an older version and comparing the listed rules in the README vs what we have in our config.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
